src/sqlfluff/core/rules/doc_decorators.py

```diff
@@ -60,12 +69,16 @@ def document_configuration(cls, ruleset="std"):
         return cls
     # Add final blank line
     config_doc += "\n"
-    # Add the configuration section immediately after the class description
-    # docstring by inserting after the first line break, or first period,
-    # if there is no line break.
-    end_of_class_description = "." if "\n" not in cls.__doc__ else "\n"
 
-    cls.__doc__ = cls.__doc__.replace(end_of_class_description, "\n" + config_doc, 1)
+    if "**Anti-pattern**" in cls.__doc__:
```

Comment:
If I understand correctly, this code reorders parts of the hand-written rule documentation. Could we instead change this to **detect** and **force** rule developers to write the docs in the desired order? I think that would be better, because I worry this code become more complex or fragile over time.

Reply:
It not reorder hand-written rule documentation, it is about doc decorators `document_fix_compatible` and `document_configuration` place to hand-written rule documentation.
when we add decorator to rule:
* Both `document_fix_compatible` and `document_configuration` will insert the first line of docstring.
* But when we use different order of `document_fix_compatible` and `document_configuration`, it would insert fix flag and config in different place.

What more, this patch also change the place we inserted.
* Before that, we insert in the end of first line

```markdown
The first line docstring
<--- the place we will insert decorators
Addition information about first line

example
```

* After that, we insert before the example
```markdown
The first line docstring

Addition information about first line

<--- the place we will insert decorators

example
```

Reply:
Agreed. It's not re-ordering and it was always inserting these two extra pieces. Just being a bit more complicated about where to insert them now.